scaffold/convert/utils.template.js

```diff
@@ -11,6 +11,30 @@ const replaceVars = (templateString, bundle, result) => {
   return _.template(templateString, options)(values);
 };
 
+// Check if an object is a Promise
+const isPromise = obj => {
+  return typeof obj === 'object' && typeof obj.then === 'function';
+};
+
+// Explicitly run App.beforeRequest middlewares in the app code. Only necessary
+// for WB scripting methods that send HTTP requests themselves, such as
+// KEY_poll, KEY_search, KEY_write, KEY_read_resource, KEY_custom_action_fields
+// and KEY_custom_search_fields.
+const runBeforeMiddlewares = (request, z, bundle) => {
+  const app = require('./');
+
+  const befores = app.beforeRequest ? app.beforeRequest : [];
+  return befores.reduce((prevResult, before) => {
+    if (!isPromise(prevResult)) {
```

Comment:
Minor, but I think you can skip the check and always wrap it in `Promise.resolve`.

```js
Promise.resolve(Promise.resolve(1)).then(i => console.log(i))
```